app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,71 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.all
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show
+    @recipe = Recipe.includes(:recipe_foods).find(params[:id])
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = current_user.recipes.new
+  end
+
+  # GET /recipes/1/edit
+  def edit; end
+
+  # POST /recipes or /recipes.json
+  def create
+    @recipe = current_user.recipes.new(recipe_params)
+
+    respond_to do |format|
+      if @recipe.save
+        format.html { redirect_to @recipe, notice: 'Recipe was successfully created.' }
+        format.json { render :show, status: :created, location: @recipe }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /recipes/1 or /recipes/1.json
+  def update
+    respond_to do |format|
+      if @recipe.update(recipe_params)
+        format.html { redirect_to @recipe, notice: 'Recipe was successfully updated.' }
+        format.json { render :show, status: :ok, location: @recipe }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /recipes/1 or /recipes/1.json
+  def destroy
+    @recipe.destroy
+    respond_to do |format|
+      format.html { redirect_to recipes_url, notice: 'Recipe was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
```

Comment:
- [ ] When I try to delete or destroy one of the recipes, I got this error. Please fix it.

![rails13](https://user-images.githubusercontent.com/55635977/142575949-34dd2762-ca8f-4ed1-9f07-1f7d3344cb14.png)

Reply:
Fixed